Enhancement: Jetpack Connect Landing Page (within wp-admin)

jeffgolenski opened the pull request:
Enhancement: Jetpack Connect Landing Page

Created a long-form landing page for the Jetpack Connect Screen within wp-admin.

![jp-landing](https://cloud.githubusercontent.com/assets/214813/14150330/8f307ca6-f676-11e5-99dd-60e7a4bd4781.png)


Submitting a PR for this to get the ball rolling. @richardmuscat take a look at the language and alter what you see fit.Then get those translations rocking out.

@zinigor @dereksmart take a look at admin-page.php and review the little bit of php I wrote, please. I'm pretty rusty with that language!

In the meantime, I've got a few design changes to make, but I don't want to hold back any language edits that need to be made for beta. I'll work on those over the next few days.

If you can, please test on different devices.

Note: There's one known issue in IE 11 where the background image stacking is a little awry with the clouds and graph / shield. Working on a fix for that!

cc @beaulebens

richardmuscat commented:
@jeffgolenski -- @jessefriedman and I spent a couple of hours going through this and just committed some changes. They're largely around text which is probably not controversial.

But wanted to elaborate on a couple of things:
- We removed a lot of text, largely sub-headings and other lines of text we felt were repetitive
- We removed the mid-page CTA and the additional text above the bottom CTA. Two CTAs are enough and if you've read through the page you don't need the extra lines of text
- I removed the <a> tags from around the images. (I think this was an error.)
- We split out the Security, Photon, and Support sections into their own boxes but kept Traffic and Stats together as they are very related
- We removed all the screenshots for two reasons: a) screenshots of Jetpack in this context aren't 'cool' enough and were making the page very hard to scan. illustrations would be great but will take too much time, and b) without them the page length is more manageable while still looking cool with the illustrations you produced
- Removed the border/background from the CTA boxes to make them stand out a bit more

richardmuscat commented:
@jeffgolenski -- re the image you're using for Photon -- do we have publishing rights for that?

richardmuscat commented:
@jeffgolenski -- finally one request. the top part of the page right now looks really green:

<img width="1275" alt="greeeeen" src="https://cloud.githubusercontent.com/assets/2287740/14165791/6da69e42-f6d2-11e5-97f6-22d15ebc7b75.png">

can we make the stats bars blue or orange or something?

jeherve commented on views/admin/admin-page.php:
```diff
@@ -240,7 +241,103 @@
+				</header>
+
+				<div class="j-feature-img">
+					<img srcset="<?php echo plugins_url( 'images/connection-landing/feature-photon-sm.jpg', JETPACK__PLUGIN_FILE ); ?> 1x,
```
How about using Photon to serve those images? It would be relevant here :)

You can use `jetpack_photon_url()` to use Photon here. It accepts 2 parameters: the image URL, and optional image parameters (size, filters, [...](https://developer.wordpress.com/docs/photon/))

richardmuscat replied:
Will it work if Jetpack is not connected though? This landing page pre-authorisation

jeffgolenski commented:
@richardmuscat 

Thanks for taking a stab at the language and stuff! 

Just a couple of friendly rebuttal points from me!

![jetpack-landing-stripped](https://cloud.githubusercontent.com/assets/214813/14181035/73b76a26-f733-11e5-805e-09e56cc1eee7.png)


> We removed a lot of text, largely sub-headings and other lines of text we felt were repetitive.

Cool.

> We removed the mid-page CTA and the additional text above the bottom CTA. Two CTAs are enough and if you've read through the page you don't need the extra lines of text.

Deal.

> I removed the <a> tags from around the images. (I think this was an error.)

anchor tags were intentional to allow people to see the details of the images on smaller screens – since a lot of the images were actual jetpack UI (ie. sharing buttons), I thought it would be nice if people could zoom in to see the details.

> We removed all the screenshots for two reasons: a) screenshots of Jetpack in this context aren't 'cool' enough and were making the page very hard to scan. illustrations would be great but will take too much time, and b) without them the page length is more manageable while still looking cool with the illustrations you produced.

Personally, I disagree. A lot of folks are visual people and scanning images with assistive text is much easier than just reading blocks on blocks on blocks of text. The illustrations I've added don't really add as much value as the screenshots did, since they are supplementary and decorative. If anything, I'd opt to put these back in and test them. 

As an example, someone who might already have a sharing button plugin might read the sharing button text and say "oh, I don't need this, I already have a plugin to do this." Whereas, if they see a screenshot, they might be prone to see it and say "I already have a sharing button plugin, but this looks much nicer, AND I can see who likes my posts! I'll use Jetpack instead."  – without a screenshot the descriptions aren't as powerful.


> We split out the Security, Photon, and Support sections into their own boxes but kept Traffic and Stats together as they are very related.
> Removed the border/background from the CTA boxes to make them stand out a bit more

Do what you've got to do! But just keep in mind, I designed the overall new landing page to be transitional for 4.0 – so the card design here matches the card designs we'll be using in 4.0. Your edited changed that around a bit. Not a huge deal, but just something to keep in mind... when we release 4.0, we'll need to make sure they're all consistent!

>  re the image you're using for Photon -- do we have publishing rights for that?

Yes, of course! What kind of a designer do you think I am, dude?! Haha. ;-) All the imagery I use comes from Unsplash (https://unsplash.com/license) – we can do anything with them, including for commercial work. 

> can we make the stats bars blue or orange or something?

Valid point. I'll make them blue, like the actual stat bars

I also see you removed all the references to WordPress.com – once we get this page rolling, can we test out some of the WordPress.com visuals and verbiage I originally used – per my theory... I think it would beneficial.

That's all! Carry on.

jeffgolenski commented:
@jeherve 

> How about using Photon to serve those images? It would be relevant here :)

You can use jetpack_photon_url() to use Photon here. It accepts 2 parameters: the image URL, and optional image parameters (size, filters, ...)

Yeah, I was actually asking @beaulebens about that. I would love to serve up our images in the plugin via photon itself... but he mentioned something about possibly being against wporg plugin guidelines?

jeherve commented:
> but he mentioned something about possibly being against wporg plugin guidelines?

Oh. I didn't think about that. It could be, I don't know.

jessefriedman commented:
@jeffgolenski and I spoke about my suggestion to remove the screenshots. Right now the screenshots don't support the message as well as they could. Jetpack doesn't really lend itself to great screenshots at the moment (4.0 will change this) and there is a lack of context when screenshots are that small. So for now we agreed to go with no images and later explore using illustrations to strengthen the message.

jeffgolenski commented:
@eliorivero It took me a bit, but I had to rebuild a bunch of elements and redo the scss for stuff. Should be good to go on the design / front-end!

eliorivero commented:
@jeffgolenski thanks for the hard work! :+1:

samhotchkiss commented:
Looks great-- I think this is ready to merge when you're ready, @eliorivero

zinigor commented:
Awesome job @jeffgolenski ! I have changed some lines to be more readable in the code, preserving everything on the client side. Also, I have added context strings for image alternative text. Otherwise it looks great. @eliorivero can you please review my latest commit? If it's good, feel free to merge.
